antiope-sns/src/Antiope/SNS.hs

```diff
@@ -1,22 +1,31 @@
 module Antiope.SNS
 ( MonadAWS
-, publishMsg
-, subscribeMsg
+, MessageId (..)
+, Protocol (..)
+, SubscriptionArn (..)
+, publishMessage
+, subscribeMessage
 , module Network.AWS.SNS
 ) where
 
 import Control.Lens
-import Data.Text       (Text)
-import Network.AWS     (MonadAWS, send)
+import Data.Text             (Text)
+import Network.AWS           (MonadAWS, send)
+import Network.AWS.Data.Text (FromText (..), ToText (..))
 import Network.AWS.SNS
 
-publishMsg :: MonadAWS m => Text -> Text -> m ()
-publishMsg topicArn message = do
-  _ <- send $ publish message & pTopicARN ?~ topicArn
-  return ()
+newtype MessageId = MessageId Text deriving (Show, Eq, Ord, ToText, FromText)
+newtype Protocol = Protocol Text deriving (Show, Eq, Ord, ToText, FromText)
+newtype SubscriptionArn = SubscriptionArn Text deriving (Show, Eq, Ord, ToText, FromText)
 
-subscribeMsg :: MonadAWS m => Text -> Text -> Text -> m ()
-subscribeMsg topicArn protocol ep = do
-  _ <- send $ subscribe topicArn protocol & subEndpoint ?~ ep
-  return ()
+publishMessage :: MonadAWS m => Topic -> Text -> m (Maybe MessageId)
+publishMessage topicArn message = do
+  resp <- send $ publish message & pTopicARN <>~ (topicArn ^. tTopicARN)
+  return $ MessageId <$> resp ^. prsMessageId
 
+subscribeMessage :: MonadAWS m => Topic -> Protocol -> Endpoint -> m (Maybe SubscriptionArn)
```

Comment:
Still, can we name it `subscribeTopic` or similar? It has `Message` in name, but it has nothing to do with messages.

Reply:
Fixed, thanks.